### PR TITLE
fix: check subscription when relay publish message

### PR DESCRIPTION
**Pull request by kaichaosun**
# Description

Relates to: https://github.com/waku-org/go-waku/issues/1207

# Changes

- [ ] ...
- [ ] ...

# Tests

### CODE REVIEW

File: waku/v2/protocol/lightpush/waku_lightpush_test.go [hunk 1]
```diff
@@ -336,7 +336,7 @@ func TestWakuLightPushCornerCases(t *testing.T) {
 
 	// Test corner case with default pubSub topic
 	_, err = client.Publish(ctx, msg2, WithDefaultPubsubTopic(), WithPeer(host2.ID()))
-	require.NoError(t, err)
+	require.Error(t, err)
```
Comment (kaichaosun) [inline waku/v2/protocol/lightpush/waku_lightpush_test.go:339]: I'm wondering if this is the right fix for this test?

Comment (richard-ramos) [reply waku/v2/protocol/lightpush/waku_lightpush_test.go:339]: I guess a way to make sure the test is correct could be to check if the error emitted is really a `errors.New("cannot publish to unsubscribed topic")`?

**Review by richard-ramos — APPROVED** [review]

File: waku/v2/protocol/relay/waku_relay.go [hunk 2]
```diff
@@ -280,12 +280,20 @@ func (w *WakuRelay) Publish(ctx context.Context, message *pb.WakuMessage, opts .
 		if err != nil {
 			return pb.MessageHash{}, err
 		}
+		_, err = w.subscribeToPubsubTopic(params.pubsubTopic)
```
Comment (chaitanyaprem) [inline waku/v2/protocol/relay/waku_relay.go:283]: why subscribe only in case pubsubtopic is empty?
shouldn't we return errro in this case as well simliar to line 294

Comment (kaichaosun) [reply waku/v2/protocol/relay/waku_relay.go:283]: For relay REST API, it supports publish a message without pubsubTopic, this is a quick fix for it.
Not sure if there's a better way for this scenario.
